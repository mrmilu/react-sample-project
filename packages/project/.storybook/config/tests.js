import { addDecorator } from '@storybook/react';
import { withTests } from '@storybook/addon-jest';

import results from '../../src/stories/.jest-test-results.json';

addDecorator(withTests({ results }));
