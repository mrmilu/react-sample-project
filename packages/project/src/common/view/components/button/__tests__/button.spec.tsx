import React from 'react';
import { customRender, mockComponent } from '../../../../../../tests/test-utils';
import Button from '../button.component';

jest.mock('../../title/title.component', () => mockComponent('Title'));

describe('Button', () => {
  it('Render', () => {
    const rendered = customRender(<Button>Sample button</Button>);
    const text = rendered.getByText(/Sample button/);
    expect(text).not.toBeNil();
    // Generate a master snapshot or check if already exists a master
    expect(rendered.container).toMatchSnapshot();
  });
  it('Should on click', () => {
    const clickEvent = jest.fn();
    const rendered = customRender(<Button onClick={clickEvent}>the button</Button>);
    rendered.getByText(/the button/).click();
    // Check if event is called 1 time and without parameters
    expect(clickEvent).toHaveBeenCalledTimes(1).toHaveBeenCalledWith();
  });
  it('Should do not click on disabled', () => {
    const clickEvent = jest.fn();
    const rendered = customRender(
      <Button disabled onClick={clickEvent}>
        the button
      </Button>
    );
    rendered.getByText(/the button/).click();
    // Check if event is not called
    expect(clickEvent).toHaveBeenCalledTimes(0);
    // or
    expect(clickEvent).not.toHaveBeenCalled();
  });
});
