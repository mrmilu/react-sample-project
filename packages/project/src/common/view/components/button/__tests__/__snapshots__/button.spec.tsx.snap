// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render 1`] = `
<div>
  <button>
    <div
      data-mocked-component="Title"
    >
      This is a button
    </div>
    Sample button
  </button>
</div>
`;
